Sources/UIKitBackend/UIViewRepresentable.swift

```diff
@@ -0,0 +1,198 @@
+    /// - Note: This may be called even when `context` has not changed.
+    func updateUIView(_ uiView: UIViewType, context: UIViewRepresentableContext<Coordinator>)
+
+    /// Make the coordinator for this view.
+    ///
+    /// The coordinator is used when the view needs to communicate changes to the rest of
+    /// the view hierarchy (i.e. through bindings), and is often the view's delegate.
+    func makeCoordinator() -> Coordinator
+
+    /// Compute the view's size.
+    /// - Parameters:
+    ///   - proposal: The proposed frame for the view to render in.
+    ///   - uiVIew: The view being queried for its preferred size.
+    ///   - context: The context, including the coordinator and environment values.
+    /// - Returns: Information about the view's size. The ``SwiftCrossUI/ViewSize/size``
+    /// property is what frame the view will actually be rendered with if the current layout
+    /// pass is not a dry run, while the other properties are used to inform the layout engine
+    /// how big or small the view can be. Pass `nil` for the maximum width/height if the view
+    /// has no maximum size (and therefore may occupy the entire screen).
+    ///
+    /// The default implementation uses `uiView.intrinsicContentSize` and `uiView.sizeThatFits(_:)`
+    /// to determine the return value.
+    func determineViewSize(
+        for proposal: SIMD2<Int>, uiView: UIViewType,
+        context: UIViewRepresentableContext<Coordinator>
+    ) -> ViewSize
+
+    /// Called to clean up the view when it's removed.
+    /// - Parameters:
+    ///   - uiVIew: The view being dismantled.
+    ///   - coordinator: The coordinator.
+    ///
+    /// This method is called after all UIKit lifecycle methods, such as
+    /// `uiView.didMoveToSuperview()`.
+    ///
+    /// The default implementation does nothing.
+    static func dismantleUIView(_ uiView: UIViewType, coordinator: Coordinator)
+}
+
+extension UIViewRepresentable {
+    public static func dismantleUIView(_: UIViewType, coordinator _: Coordinator) {
+        // no-op
+    }
+
+    public func determineViewSize(
+        for proposal: SIMD2<Int>, uiView: UIViewType,
+        context _: UIViewRepresentableContext<Coordinator>
+    ) -> ViewSize {
+        let intrinsicSize = uiView.intrinsicContentSize
+        let sizeThatFits = uiView.sizeThatFits(
+            CGSize(width: CGFloat(proposal.x), height: CGFloat(proposal.y)))
+        let roundedSize = SIMD2(
+            Int(sizeThatFits.width.rounded(.up)),
+            Int(sizeThatFits.height.rounded(.up)))
+        return ViewSize(
+            size: SIMD2(
+                intrinsicSize.width < 0.0 ? proposal.x : min(proposal.x, roundedSize.x),
+                intrinsicSize.height < 0.0 ? proposal.y : min(proposal.y, roundedSize.y)
+            ),
+            idealSize: SIMD2(
+                intrinsicSize.width < 0.0 ? proposal.x : roundedSize.x,
+                intrinsicSize.height < 0.0 ? proposal.y : roundedSize.y
```

Comment:
Ideal size should be 10 when intrinsic size is negative and equal to the intrinsic size otherwise.

By convention, SwiftUI and SwiftCrossUI use 10 for the default ideal size along each dimension when a view can take on any size (e.g. `Color`). Some views such as a resizable `Image` can take on any size but instead use the dimensions of the underlying image as their ideal size, but in this default implementation we can't really know what the underlying sizing behaviour of the `UIView` is, so 10 is probably the best we can do.

The idea behind `idealSize` is that it should always have the same value for the same content, regardless of proposed size. It's required for things like the `fixedSize()` modifier (which SwiftCrossUI doesn't have yet).